gammapy/utils/units.py

```diff
@@ -102,9 +102,12 @@ def energy_unit_format(E):
         Returns a string or list of strings with energy unit formatted        
     """   
 
-    if not np.shape(E):  # scalar
-        return energy_str_formatting(E)
-    
-    E_fmt = [energy_str_formatting(e) for e in E]
-    return E_fmt
-    
\ No newline at end of file
+    if (isinstance(E, (list, tuple)) == True):
```

Comment:
This is the not easy part.
the input can be different nature:
- astropy quantity (single Q or array of Q)
- list or tuple of Q
I couldn't find a way to simply disentangle between iterable and non iterable inputs.
size, shape, or len doesn't work on  all inputs types.